### PR TITLE
AO3-5062 Enable work download feature tests

### PR DESCRIPTION
**Pull request by redsummernight**
## Issue

https://otwarchive.atlassian.net/browse/AO3-5062

## Purpose

Install wkhtmltopdf and html2mobi on Travis and Codeship.
In work_download.feature, check downloaded file names and types.

**Note**: the Codeship test pipeline needs one extra step for setting the environment variable `PERL5LIB` because I can't do it from within a script. The wiki should be updated similarly.

```sh
export RAILS_ENV=test
export PERL5LIB="${HOME}/perl5/lib/perl5"
bash ./script/prepare_codeship.sh
# ...
```

The source for html2mobi was from here: http://www.ida.liu.se/~tompe44/mobiperl/.
## Testing

None, automated.

### CODE REVIEW

File: script/travis_ebook_converters.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+cpanm install XML::Parser::Lite::Tree --sudo -q
+cpanm install Image::BMP --sudo -q
+cpanm install Image::Size --sudo -q
+cpanm install HTML::TreeBuilder --sudo -q
+cpanm install Getopt::Mixed --sudo -q
+cpanm install Date::Parse --sudo -q
+cpanm install Date::Format --sudo -q
+cpanm install URI::Escape --sudo -q
+
+tar xvf script/mobiperl-0.0.43.tar.gz
```
Comment (zz9pzza) [inline script/travis_ebook_converters.sh]: Can we do the wget thing so we don't have to include the source to mobiperl in this pull ?

**Comment by zz9pzza** [comment]
Can you get this working on codeship as well ?

**Comment by zz9pzza** [comment]
( Ps I do like it :) )

**Comment by redsummernight** [comment]
@zz9pzza I overlooked Codeship. :<

I can install wkhtmltopdf like this:
```sh
wget https://github.com/wkhtmltopdf/wkhtmltopdf/releases/download/0.12.4/wkhtmltox-0.12.4_linux-generic-amd64.tar.xz
tar --directory="${HOME}" -xvf wkhtmltox-0.12.4_linux-generic-amd64.tar.xz
ln -s "${HOME}/wkhtmltox/bin/wkhtmltopdf" "${HOME}/bin/wkhtmltopdf"
wkhtmltopdf -h
```

I can [install cpanm](http://cwinters.com/2015/10/03/recipe-for-sqitch-on-codeship.html).

Codeship [does not allow running sudo commands](https://documentation.codeship.com/basic/builds-and-configuration/root-level-access/), which means I can't install libgd2-xpm-dev, which is a requirement of GD, which is a dependency of MobiPerl. I'm not sure what to do, should we contact Codeship support?

When we switch to calibre though, Codeship has it [installed by default](https://documentation.codeship.com/basic/builds-and-configuration/packages/).

**Comment by zz9pzza** [comment]
Looking at https://community.codeship.com/t/apt-get-install-graphicsmagick/139

"Codeship already has graphicsmagick installed"  ?

**Comment by redsummernight** [comment]
Download converters now work on Codeship. Kudos to Codeship's support, they sent me the script for installing libgd2, one of mobiperl's dependencies.

**Note**: the Codeship test pipeline needs one extra step for setting the environment variable `PERL5LIB` because I can't do it from within a script.

```sh
export RAILS_ENV=test
export PERL5LIB="${HOME}/perl5/lib/perl5"
bash ./script/prepare_codeship.sh
# ...
```

Coverage for downloads_controller increases from 34.19 to 91.45: https://coveralls.io/builds/12411565/source?filename=app%2Fcontrollers%2Fdownloads_controller.rb

**Comment by sarken** [comment]
Pinging @zz9pzza to see if this is ready

**Comment by redsummernight** [comment]
Please let me know if I need to change the version of either converters to match what AO3 is using on non-test environments.

**Comment by redsummernight** [comment]
Replaced by #3392.

**Pull request closed without merging**